Storage/Data/CapturedCarService.cs

```diff
@@ -61,16 +61,19 @@ public void Add(CapturedCar car)
         public void RemoveByCarId(int id)
         {
             Car car = carData.FindById(id);
-            List<Image> images = imageData.FindAll();
+            if (car != null)
+            {
+                List<Image> images = imageData.FindAll();
 
-            IEnumerable<Image> imgs = from image in images
-                                      where car.Id == image.CarId
-                                      select image;
+                IEnumerable<Image> imgs = from image in images
+                                          where car.Id == image.CarId
+                                          select image;
 
-            carData.Remove(car.Id);
-            foreach (var img in imgs)
-            {
-                imageData.Remove(img.Id);
+                carData.Remove(car.Id);
```

Comment:
Seems like these find and remove, should be implemented into imageData, single responsibility